packages/core/src/plugin/rewrite-request.ts

```diff
@@ -1,29 +1,47 @@
+import path from 'node:path';
 import type { ConfigT } from 'metro-config';
-import type { ModuleFederationConfigNormalized } from '../types';
 import { MANIFEST_FILENAME } from './constants';
-import { replaceExtension } from './helpers';
+import { removeExtension } from './helpers';
 
 type CreateRewriteRequestOptions = {
-  options: ModuleFederationConfigNormalized;
   config: ConfigT;
+  originalEntryFilename: string;
+  remoteEntryFilename: string;
   manifestPath: string;
+  tmpDirPath: string;
 };
 
 export function createRewriteRequest({
-  options,
   config,
+  originalEntryFilename,
+  remoteEntryFilename,
   manifestPath,
+  tmpDirPath,
 }: CreateRewriteRequestOptions) {
+  const hostEntryName = removeExtension(originalEntryFilename);
+  const remoteEntryName = removeExtension(remoteEntryFilename);
+  const relativeTmpDirPath = path
+    .relative(config.projectRoot, tmpDirPath)
+    .split(path.sep)
+    .join(path.posix.sep);
+  const hostEntryPathRegex = getEntryPathRegex(hostEntryName);
+  const remoteEntryPathRegex = getEntryPathRegex(remoteEntryName);
+
   return function rewriteRequest(url: string) {
+    const root = config.projectRoot;
     const { pathname } = new URL(url, 'protocol://host');
-    // rewrite /mini.bundle -> /mini.js.bundle
-    if (pathname.startsWith(`/${options.filename}`)) {
```

Comment:
Seems like you deleted quite important logic here. Is it no longer required? I think it might break usecases where we use custom filenames in remotes object.